src/app/_models/question_counts.ts

```diff
@@ -0,0 +1,9 @@
+export interface QuestionCount{
+    name: string;
+    count: number;
+    count_per_difficulty: DifficultyCount[];
+}
+interface DifficultyCount{
```

Comment:
There is no need to have DifficultyCount as a separate interface please have it inline in the QuestionCount interface.
```
export interface QuestionCount{
    name: string;
    count: number;
    count_per_difficulty: {count: number; difficulty: string}[];
}
```

Reply:
Did exactly this.